File: 2041-lab-files/lab06/whale.py
#!/usr/bin/env python3
# let's do this one smartly (only one arg allowed by spec)
import sys
# fileinput.input() works like Perl's empty <>
# Python 2: raw_input(), Python 3: input()
# sys.stdin.read() or .readlines()

# note: assignments not allowed in expressions in Python
total = 0
n_pods = 0
for line in sys.stdin: # best method
    line = line.rstrip('\n')
    n_whales, whale_type = line.split(' ', 1)
    # 1: means, num splits to be made
    n_whales = int(n_whales)
    if whale_type == sys.argv[1]:
        n_pods += 1
        total += n_whales
print(sys.argv[1], "observations:", n_pods, "pods,",
        total, "individuals")

File: 2041-lab-files/lab06/whale.pl
#!/usr/bin/perl

# read from stdin
# either make use of two hashes, or a multi-dimensional hash
my %whale_pods;
while ($line = <STDIN>) {
    chomp $line;
    ($num, $type) = split(/ /, $line, 2);
    $whale_pods{$type}{"n_pods"} += 1;
    $whale_pods{$type}{"total"} += $num;
}
# print contents for that whale name in ARGV[0]
#   (Why are you keeping track of all them when you
#   only want a single whale???)
$key = $ARGV[0];
# if key doesn't exist, the values eq undef
if (defined $whale_pods{"$key"}) {
    $n_pods = $whale_pods{"$key"}{"n_pods"};
    $total = $whale_pods{"$key"}{"total"};
} else {
    $n_pods = 0;
    $total = 0;
}
print "$key observations: $n_pods pods, $total individuals\n";


# just prints out everything
<<"COMMENT"
foreach $key (keys %whale_pods) {
    $n_pods = $whale_pods{"$key"}{"n_pods"};
    $total = $whale_pods{"$key"}{"total"};
    print "$key observations: $n_pods pods, $total individuals\n";
}
COMMENT
